File: rxanimation/src/main/java/com/reyurnible/rxanimation/animation/RxAnimation.java
package com.reyurnible.rxanimation.animation;

import io.reactivex.Observable;
import io.reactivex.ObservableOnSubscribe;

import android.animation.ObjectAnimator;
import android.support.annotation.CheckResult;
import android.support.annotation.NonNull;
import android.view.View;
import android.view.animation.Animation;

/**
 * Static factory methods for animation event observable.
 */
public final class RxAnimation {

    /**
     * Create an animation event observable.
     */
    @CheckResult
    @NonNull
    public static Observable<AnimationEvent> events(@NonNull Animation animation, @NonNull View view) {
        return Observable.create(new AnimationEventOnSubscribe(animation, view));
    }
}

File: rxanimation-kotlin/src/main/kotlin/com/reyurnible/rxanimation_kotlin/animation/RxAnimation.kt
package com.reyurnible.rxanimation_kotlin.animation

import android.view.View
import android.view.animation.Animation
import com.reyurnible.rxanimation.animation.AnimationEvent
import com.reyurnible.rxanimation.animation.RxAnimation
import io.reactivex.Observable

fun Animation.events(view: View): Observable<AnimationEvent> = RxAnimation.events(this, view)
